lua_cjson.c

```diff
@@ -1240,7 +1248,10 @@ static void json_process_value(lua_State *l, json_parse_t *json,
     case T_STRING:
         lua_pushlstring(l, token->value.string, token->string_len);
         break;;
-    case T_NUMBER:
+    case T_INTEGER:
+        lua_pushinteger(l, token->value.number);
+        break;;
+    case T_DOUBLE:
```

Comment:
Shouldn't this still be named "number", as the number type is generic, and may be 32 or 64 bit?
(i.e. it's called `lua_pushnumber` for that reason)